libcrux-ml-dsa/src/impl_signature_trait.rs

```diff
@@ -0,0 +1,131 @@
+#[cfg(not(eurydice))]
+#[cfg_attr(hax, hax_lib::exclude)]
+pub mod signers {
+    //! [`libcrux_traits::signature`] APIs.
+    use libcrux_traits::signature::owned;
+
+    /// A trait representing the context used for ML-DSA signing and verification.
+    ///
+    /// Can be implemented using the convenience macro [`impl_context`].
+    pub trait Context {
+        /// Return the context.
+        fn context() -> &'static [u8];
+    }
+
+    #[macro_export]
+    /// A convenience macro for implementing the [`Context`] trait.
+    /// The `$context` must be provided as a `&'static [u8]`.
+    ///
+    /// Usage:
+    /// ```rust
+    /// use libcrux_ml_dsa::signers::{Context, impl_context};
+    /// impl_context!(AppContext, b"context");
+    /// ```
+    macro_rules! impl_context {
+        ($name:ident, $context:expr) => {
```

Comment:
Maybe this should have a way for the user to provide a comment to the type that we define? Or we just assign one based on the byte string.

Reply:
If we do it based on the context, we might want to make it a literal, though.